do not allow games where the secret santa can be deterministically deduced

jcken95 opened the pull request:
`gift_recipient = random_names[c(2:length(random_names), 1)]` will fail if `length(names) <= 1`, however, the secret santa is meaningless for 2 or fewer players, so we should error

jcken95 committed:
fix: prevent non-secret games

jcken95 committed:
feat: test error handling for non-secret games

jcken95 commented:
banging this into your PR to save nicola some effort

jcken95 committed:
chore: lint

jcken95 committed:
fix: 4 is the minimum number for a truely secret SS

keithnewman commented on tests/testthat/test-secret_santa.R:
```diff
@@ -22,3 +22,11 @@ test_that("secret_santa() warns if the same person appears twice in the list", {
+test_that("secret_santa() warns if the number of players is too small", {
+  nice_list <- c(
+    "Rudolph",
+    "Dasher"
```
Just adding a third player in
```suggestion
    "Dasher",
    "Prancer"
```

keithnewman reviewed (changes requested):
Very good point, Jack. I am going to squash your commits though

jcken95 committed:
chore: update the nice list

apparently i am being unfair, but it is very unfair that prancer ate all the carrots last year :shrug:

Co-authored-by: Keith Newman <keithnewman@users.noreply.github.com>

keithnewman reviewed (approved):
Excellent work @jcken95
